### DISCUSSION TITLE
Request to change the default EnforcedStyle for Layout/DotPosition

### DISCUSSION THREAD

**Discussion by pioz** [rubocop/rubocop#12028] [General]
The default style for the [Layout/DotPosition](https://www.rubydoc.info/gems/rubocop/RuboCop/Cop/Layout/DotPosition) cop in RuboCop is currently set to "leading". The alternative option is "trailing".

While the default "leading" style is acceptable, I've encountered a significant drawback that affects its usability.
```ruby
product = Product
  .where(public: true)
  .order(created_at: :desc)
```

When code is written in this style and copied into the console, an error is thrown:
`SyntaxError ((irb):2: syntax error, unexpected '.')`.

On the other hand, the "trailing" style offers a solution to this issue:
```ruby
product = Product.
  where(public: true).
  order(created_at: :desc)
```
Using this style allows the code to be copied and pasted into the console seamlessly.

Considering the utility of use, I believe the "trailing" style is better and should be the default choice for RuboCop.

I would appreciate hearing your thoughts on this matter.

**Comment by bbatsov** [comment]
We are unlike to ever change this default, given that the leading style is a lot more popular in the Ruby community these days. Both styles have some merits and some issues as discussed in great detail 10 years ago https://github.com/rubocop/ruby-style-guide/pull/176
